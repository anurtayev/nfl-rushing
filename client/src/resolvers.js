import gql from "graphql-tag";

export const typeDefs = gql`
  extend type Query {
    sortBy: String!
    filter: String
    topCursor: Int
    bottomCursor: Int
  }
`;

export const resolvers = {};
