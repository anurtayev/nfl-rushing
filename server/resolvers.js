export default {
  Query: {
    entries: async (
      _,
      { sortBy, filter, pageSize, cursor, direction },
      { dataSources }
    ) => {
      return await dataSources.rushingAPI.getEntries({
        sortBy,
        filter,
        pageSize,
        cursor,
        direction
      });
    },

    csv: async (_, { sortBy, filter }, { dataSources }) => {
      return await dataSources.rushingAPI.getCsv({
        sortBy,
        filter
      });
    }
  }
};
