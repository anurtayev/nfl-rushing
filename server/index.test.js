import { ApolloServer } from "apollo-server";
import typeDefs from "./schema";
import resolvers from "./resolvers";
import gql from "graphql-tag";
import rushingAPI from "./datasources/rushing";
import { createTestClient } from "apollo-server-testing";

const dataSources = () => ({ rushingAPI });
const server = new ApolloServer({
  typeDefs,
  resolvers,
  dataSources,
  introspection: true,
  playground: true
});

const TEST_QUERY = gql`
  {
    entries {
      id
      player
      team
      pos
      attg
      att
      yds
      avg
      ydsg
      td
      lng
      first
      firstPercentage
      twentyPlus
      fortyPlus
      fum
    }
  }
`;

describe("ApolloServer", () => {
  it("should return fetch data correctly", async () => {
    const { query } = createTestClient(server);
    const res = await query({ query: TEST_QUERY });
    expect(res).toMatchSnapshot();
  });
});
