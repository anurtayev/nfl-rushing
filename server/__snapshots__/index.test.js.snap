// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApolloServer should return fetch data correctly 1`] = `
Object {
  "data": Object {
    "entries": Array [
      Object {
        "att": 2,
        "attg": 0.1,
        "avg": -11.5,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "0",
        "lng": "0",
        "player": "Sam Koch",
        "pos": "P",
        "td": 0,
        "team": "BAL",
        "twentyPlus": 0,
        "yds": -23,
        "ydsg": -1.4,
      },
      Object {
        "att": 2,
        "attg": 1,
        "avg": -6,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "1",
        "lng": "0",
        "player": "DeMarcus Ayers",
        "pos": "WR",
        "td": 0,
        "team": "PIT",
        "twentyPlus": 0,
        "yds": -12,
        "ydsg": -6,
      },
      Object {
        "att": 21,
        "attg": 1.3,
        "avg": -0.4,
        "first": 2,
        "firstPercentage": 9.5,
        "fortyPlus": 0,
        "fum": 0,
        "id": "2",
        "lng": "6",
        "player": "Eli Manning",
        "pos": "QB",
        "td": 0,
        "team": "NYG",
        "twentyPlus": 0,
        "yds": -9,
        "ydsg": -0.6,
      },
      Object {
        "att": 1,
        "attg": 0.1,
        "avg": -8,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "3",
        "lng": "-8",
        "player": "Taiwan Jones",
        "pos": "RB",
        "td": 0,
        "team": "OAK",
        "twentyPlus": 0,
        "yds": -8,
        "ydsg": -0.6,
      },
      Object {
        "att": 2,
        "attg": 0.2,
        "avg": -3,
        "first": 1,
        "firstPercentage": 50,
        "fortyPlus": 0,
        "fum": 0,
        "id": "4",
        "lng": "2",
        "player": "James Wright",
        "pos": "WR",
        "td": 0,
        "team": "CIN",
        "twentyPlus": 0,
        "yds": -6,
        "ydsg": -0.5,
      },
      Object {
        "att": 5,
        "attg": 1.2,
        "avg": -1.2,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "5",
        "lng": "-1",
        "player": "Ryan Mallett",
        "pos": "QB",
        "td": 0,
        "team": "BAL",
        "twentyPlus": 0,
        "yds": -6,
        "ydsg": -1.5,
      },
      Object {
        "att": 1,
        "attg": 0.1,
        "avg": -5,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "6",
        "lng": "-5",
        "player": "Travis Kelce",
        "pos": "TE",
        "td": 0,
        "team": "KC",
        "twentyPlus": 0,
        "yds": -5,
        "ydsg": -0.3,
      },
      Object {
        "att": 4,
        "attg": 1.3,
        "avg": -1,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "7",
        "lng": "-1",
        "player": "Nick Foles",
        "pos": "QB",
        "td": 0,
        "team": "KC",
        "twentyPlus": 0,
        "yds": -4,
        "ydsg": -1.3,
      },
      Object {
        "att": 6,
        "attg": 0.8,
        "avg": -0.7,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "8",
        "lng": "0",
        "player": "Landry Jones",
        "pos": "QB",
        "td": 0,
        "team": "PIT",
        "twentyPlus": 0,
        "yds": -4,
        "ydsg": -0.5,
      },
      Object {
        "att": 2,
        "attg": 2,
        "avg": -2,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "9",
        "lng": "1",
        "player": "Brandon Burks",
        "pos": "RB",
        "td": 0,
        "team": "NYJ",
        "twentyPlus": 0,
        "yds": -4,
        "ydsg": -4,
      },
      Object {
        "att": 12,
        "attg": 0.9,
        "avg": -0.3,
        "first": 2,
        "firstPercentage": 16.7,
        "fortyPlus": 0,
        "fum": 1,
        "id": "10",
        "lng": "5",
        "player": "Reggie Bush",
        "pos": "RB",
        "td": 1,
        "team": "BUF",
        "twentyPlus": 0,
        "yds": -3,
        "ydsg": -0.2,
      },
      Object {
        "att": 1,
        "attg": 0.1,
        "avg": -3,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 1,
        "id": "11",
        "lng": "-3",
        "player": "Jordan Berry",
        "pos": "P",
        "td": 0,
        "team": "PIT",
        "twentyPlus": 0,
        "yds": -3,
        "ydsg": -0.2,
      },
      Object {
        "att": 3,
        "attg": 1,
        "avg": -1,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "12",
        "lng": "-1",
        "player": "Matt McGloin",
        "pos": "QB",
        "td": 0,
        "team": "OAK",
        "twentyPlus": 0,
        "yds": -3,
        "ydsg": -1,
      },
      Object {
        "att": 2,
        "attg": 0.1,
        "avg": -1.5,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "13",
        "lng": "-1",
        "player": "Travis Benjamin",
        "pos": "WR",
        "td": 0,
        "team": "SD",
        "twentyPlus": 0,
        "yds": -3,
        "ydsg": -0.2,
      },
      Object {
        "att": 3,
        "attg": 0.6,
        "avg": -1,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "14",
        "lng": "-1",
        "player": "Drew Stanton",
        "pos": "QB",
        "td": 0,
        "team": "ARI",
        "twentyPlus": 0,
        "yds": -3,
        "ydsg": -0.6,
      },
      Object {
        "att": 1,
        "attg": 0.1,
        "avg": -3,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "15",
        "lng": "-3",
        "player": "Will Fuller",
        "pos": "WR",
        "td": 0,
        "team": "HOU",
        "twentyPlus": 0,
        "yds": -3,
        "ydsg": -0.2,
      },
      Object {
        "att": 3,
        "attg": 0.8,
        "avg": -0.7,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 1,
        "id": "16",
        "lng": "0",
        "player": "Brett Hundley",
        "pos": "QB",
        "td": 0,
        "team": "GB",
        "twentyPlus": 0,
        "yds": -2,
        "ydsg": -0.5,
      },
      Object {
        "att": 1,
        "attg": 0.1,
        "avg": -2,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "17",
        "lng": "-2",
        "player": "Brandon LaFell",
        "pos": "WR",
        "td": 0,
        "team": "CIN",
        "twentyPlus": 0,
        "yds": -2,
        "ydsg": -0.1,
      },
      Object {
        "att": 1,
        "attg": 0.1,
        "avg": -2,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "18",
        "lng": "-2",
        "player": "Valentino Blake",
        "pos": "DB",
        "td": 0,
        "team": "TEN",
        "twentyPlus": 0,
        "yds": -2,
        "ydsg": -0.1,
      },
      Object {
        "att": 2,
        "attg": 0.5,
        "avg": -1,
        "first": 0,
        "firstPercentage": 0,
        "fortyPlus": 0,
        "fum": 0,
        "id": "19",
        "lng": "-1",
        "player": "Matt Schaub",
        "pos": "QB",
        "td": 0,
        "team": "ATL",
        "twentyPlus": 0,
        "yds": -2,
        "ydsg": -0.5,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
